scripts/prepare_sector_network.py

```diff
@@ -1882,7 +1882,7 @@ def add_land_transport(n, costs):
                 carrier="land transport fuel cell",
                 p_set=fuel_cell_share
                 / options["transport_fuel_cell_efficiency"]
-                * transport[nodes],
+                * transport[[node for node in nodes if node in transport.index]],

```

Comment:
It would be highly beneficial to adopt good data standards as otherwise the code can become quite obscure and these ad-hoc solutions may explode easily.

I'm commenting this as I see likely that the same root cause will percolate and appear in many different ways in different countries and options, leading to more efforts when not tackled directly.

Agree that understanding why transport does not have the full expected buses sounds a good idea and filling them even during the clustering phase sounds quite interesting. It may also be due to the population layout that may not have full entries.
Alternatively, having a utility function that fills the index

What do you think?